cql3: add USING TIMEOUT support for deletes

psarna opened the pull request:
Turns out the DELETE statement already supports attributes
like timestamp, so it's ridiculously easy to add USING TIMEOUT
support - it's just the matter of accepting it in the grammar.

Fixes #8855

avikivity commented on cql3/Cql.g:
```diff
@@ -614,6 +614,7 @@ deleteOp returns [std::unique_ptr<cql3::operation::raw_deletion> op]
 
 usingClauseDelete[std::unique_ptr<cql3::attributes::raw>& attrs]
     : K_USING K_TIMESTAMP ts=intValue { attrs->timestamp = ts; }
+    | K_USING K_TIMEOUT to=term { attrs->timeout = to; }
```
K_UPDATE K_THE K_DOCUMENTATION

avikivity replied:
Oh, and USING TIMEOUT :timeout USING TIMESTAMP :ts should be supported.

avikivity replied:
I mean, USING TIMEOUT :timeout AND TIMESTAMP :ts, I think.

psarna replied:
ah, right

psarna commented:
v2:
 - added support for specifying multiple attrs (by simplifying the grammar along the way)
 - added extra validation for TTL - which is not a valid parameter for deletes
 - adjusted the tests

avikivity commented on cql3/statements/modification_statement.cc:
```diff
@@ -675,7 +675,11 @@ modification_statement::modification_statement(cf_name name, std::unique_ptr<att
+    if (_attrs->time_to_live) {
+        throw exceptions::invalid_request_exception("TTL attribute is not allowed for deletes");
+    }
+}
```
Is this constructor used exclusively for deletes? Its kind of awkward to put this delete-specific code here.

Maybe move to delete_statemet c'tor?

psarna replied:
aww, vscode tricked me :( I meant to put it in there

psarna commented:
v3:
 - moved the validation to the delete statement where it belongs
 - actually reran all unit tests to ensure that this validation works
